### PR TITLE
[CMAKE] Eliminate the use of GCC and CLANG variables

### PR DESCRIPTION
**Pull request by Extravert-ir**
Now, these should be used:
MSVC - for compilers with MSVC-like API
NOT MSVC - for compilers with GNU-like API

CMAKE_C_COMPILER_ID for everything else

### CODE REVIEW

File: dll/3rdparty/mbedtls/CMakeLists.txt [hunk 1]
```diff
@@ -95,6 +95,8 @@ add_cd_file(TARGET mbedtls DESTINATION reactos/system32 FOR all)
 
 if(NOT MSVC)
     target_compile_options(mbedtls PRIVATE -Wno-pointer-sign -Wno-unused-function)
-elseif(USE_CLANG_CL)
+endif()
+
+if(MAKE_C_COMPILER_ID STREQUAL "Clang")
```
Comment (andrew-boyarshin) [inline dll/3rdparty/mbedtls/CMakeLists.txt:100]: CMAKE?

File: base/services/nfsd/CMakeLists.txt [hunk 2]
```diff
@@ -48,7 +48,7 @@ list(APPEND SOURCE
 
 add_executable(nfsd ${SOURCE} nfsd.rc)
 
-if(MSVC AND (NOT USE_CLANG_CL))
+if(CMAKE_C_COMPILER_ID STREQUAL "MSVC")
```
Comment (learn-more) [inline base/services/nfsd/CMakeLists.txt:51]: strequal is case sensitive, is this always uppercase?

Comment (learn-more) [reply base/services/nfsd/CMakeLists.txt:51]: seems so.

**Comment by learn-more** [comment]
What do we gain by removing those variables and resorting to STREQUAL for the checks?

File: dll/3rdparty/mbedtls/CMakeLists.txt (hunk 1 above)
Comment (Extravert-ir) [reply dll/3rdparty/mbedtls/CMakeLists.txt:100]: Thanks!

**Comment by Extravert-ir** [comment]
These variables were confusing:
`MSVC` is a cmake built-in variable, and is set for MSVC and clang-cl
`GCC` is set for GCC and clang in GCC mode
`CLANG` is set for clang and clang-cl

So `if(GCC)` does not mean the condition is true only for GCC, which looks weird (for me at least)

I thought about introducing something like `COMPILER_API_GNU` and `COMPILER_API_MSVC`, but for our set of compilers it would basically set to `NOT MSVC` and `MSVC` respectively

**Comment by Extravert-ir** [comment]
The usage of `USE_CLANG_CL` should also be reduced btw

File: sdk/include/xdk/ntbasedef.h [hunk 3]
```diff
@@ -788,7 +788,11 @@ extern "C++" { \
 #endif /* _M_AMD64 */
 
 /* C_ASSERT Definition */
-#define C_ASSERT(expr) extern char (*c_assert(void)) [(expr) ? 1 : -1]
+#ifndef _Static_assert
+ #define C_ASSERT(expr) typedef char c_assert [(expr) ? 1 : -1]
+#else
+ #define C_ASSERT(expr) _Static_assert(expr, "(" #expr ") failed")
+#endif
```
Comment (HBelusca) [inline sdk/include/xdk/ntbasedef.h:795]: Why is this a problem with Clang? Does it fail to compile even if the static assertion is true?

Comment (Extravert-ir) [reply sdk/include/xdk/ntbasedef.h:795]: As far as I understand, it just doesn't like how we define our `C_ASSERT`

File: sdk/tools/xml2sdb/xml2sdb.cpp [hunk 4]
```diff
@@ -16,18 +16,14 @@ using tinyxml2::XMLText;
...
-C_ASSERT(sizeof(wchar_t) == 2);
-C_ASSERT(sizeof(TAG) == 2);
-C_ASSERT(sizeof(TAGID) == 4);
+static_assert(sizeof(GUID) == 16, "GUID has wrong size");
+static_assert(sizeof(ULONG) == 4, "ULONG has wrong size");
+static_assert(sizeof(LARGE_INTEGER) == 8, "LARGE_INTEGER has wrong size");
+static_assert(sizeof(WCHAR) == 2, "WCHAR has wrong size");
+static_assert(sizeof(wchar_t) == 2, "wchar_t has wrong size");
+static_assert(sizeof(TAG) == 2, "TAG has wrong size");
+static_assert(sizeof(TAGID) == 4, "TAGID has wrong size");
```
Comment (HBelusca) [inline sdk/tools/xml2sdb/xml2sdb.cpp:26]: That "has wrong size" thing repeated over & over is much copy-pasting. Maybe, for host-tools, we could have one centralized definition of C_ASSERT, exactly as done as in the XDK, and use that everywhere; that would save the copypaste effort you've done there.

File: sdk/include/xdk/ntbasedef.h (hunk 3 above)
Comment (HBelusca) [reply sdk/include/xdk/ntbasedef.h:795]: The way we do it is the same as the MS PSDK or something like that 🤔

**Pull request merged**